### PR TITLE
Big endian fixes - part 2

### PR DESCRIPTION
**Pull request by BSzili**
These were missing from the new platform code.

### CODE REVIEW

File: src/S3/s3sound.c [hunk 1]
```diff
@@ -55,10 +55,10 @@ int S3LoadSample(tS3_sound_id id) {
...
         wav_header* hdr = (wav_header*)buf;
         sample->dataptr = &buf[sizeof(wav_header)];
-        sample->size = hdr->data_bytes;
-        sample->rate = hdr->sample_rate;
-        sample->resolution = hdr->bit_depth;
-        sample->channels = hdr->num_channels;
+        sample->size = BrSwap32(hdr->data_bytes);
+        sample->rate = BrSwap32(hdr->sample_rate);
+        sample->resolution = BrSwap16(hdr->bit_depth);
+        sample->channels = BrSwap16(hdr->num_channels);
```
Comment (madebr) [inline src/S3/s3sound.c:61]: Is an `#ifdef` missing here?
```suggestion
#if BR_ENDIAN_BIG
        sample->size = BrSwap32(hdr->data_bytes);
        sample->rate = BrSwap32(hdr->sample_rate);
        sample->resolution = BrSwap16(hdr->bit_depth);
        sample->channels = BrSwap16(hdr->num_channels);
#else
        sample->size = hdr->data_bytes;
        sample->rate = hdr->sample_rate;
        sample->resolution = hdr->bit_depth;
        sample->channels = hdr->num_channels;
#endif
```

Comment (BSzili) [reply src/S3/s3sound.c:61]: Oops, indeed. I got these mixed up with BrLtoHL and BrLtoHS.

**Review by dethrace-labs — APPROVED** [review]

**Comment by Link4Electronics** [comment]
Hello, I clone your main fork and tried to compile on a PowerMac G5 running linux PPC64. During compilation it shows a lot of warnings like: warning: "BR_ENDIAN_BIG 0 <command-line>: note: this is the location of the previous definition."
It does compile either way, but when I try to run it gives this:

******************
Caught SIGSEV
******************

Stack trace:
0: BrPlaneEquation at /lib/BRender-v1.3.2/core/math/plane.c:35
1: PrepareFaceNormals at /lib/BRender-v1.3.2/core/v1db/prepmesh.c:355
2: BrModelUpdate at /lib/BRender-v1.3.2/core/v1db/prepmesh.c:1134
3: BrModelAdd at /lib/BRender-v1.3.2/core/v1db/regsupt.c:55
4: InitializeBRenderEnvironment at /src/DETHRACE/common/init.c:285
5: InitialiseApplication at /src/DETHRACE/common/init.c:377
6: InitialiseDeathRace at /src/DETHRACE/common/init.c:429
7: GameMain at /src/DETHRACE/common/main.c:122
8: original_main at /src/DETHRACE/pc-win95/win95sys.c:969
9: main at /src/DETHRACE/main.c:48
10: ?? ??:0

**Comment by BSzili** [comment]
Please don't try to build this yet for big endian, some of my are still local only.

**Comment by BSzili** [comment]
@madebr This one is good to merge, I meant the BRender repo that needs a couple header/build system changes for the BR_ENDIAN_* macros to properly propagate. Sorry for the confusion.

**Pull request merged**